cloudferrylib/os/image/glance_image.py

```diff
@@ -228,7 +232,7 @@ def delete_image(self, image_id):
         self.glance_client.images.delete(image_id)
 
     def get_image_by_id(self, image_id):
-        for glance_image in self.get_image_list():
+        for glance_image in self.get_image_list_without_filtering():
```

Comment:
Please change `convert` function accordingly: set `container_format` to `None` if it's missing in dictionary.